docs/content/docs/ops/state/checkpoints.md

```diff
@@ -175,6 +175,35 @@ above).
 $ bin/flink run -s :checkpointMetaDataPath [:runArgs]
 ```
 
+### Checkpoint/Savepoint guarantees
+Not always the snapshot can be successfully restored. 
+The following list contains operations which can be done without recovery lose depends on the type of the snapshot.
+
```

Comment:
Would it make sense to group the table rows in, e.g., (just an idea):
```
Application changes

- Job upgrade (?)
- Artibrary job upgrade (?)
- Schema evoluation

State management

- Self-contained and relocatable
- State processor API (reading)
- State processor API (writing)

Configuration changes

- State backend migration
- Minor version migration
- Bugfix/patch
- Rescaling
```

Reply:
I like the idea of grouping if these groups make sense

Reply:
Discussable. But I personally don't think that grouping makes sense here since this grouping is not natural. I mean having groups doesn't give us or users more understanding or advantages. for example, if we could say that we support all types of snapshot for group 'Application changes' but only native savepoint for group 'Configuration changes' it would be useful. But in the current implementation, it will give us only more lines in the table